src/components/MAPI/Experiments.tsx

```diff
@@ -28,6 +32,44 @@ const Experiments: React.FC<IExperimentsProps> = () => {
     }
   };
 
+  const auth = useAuthProvider();
+  const [impersonationUser, setImpersonationUser] = useState<string>('');
+  const [impersonationGroup, setImpersonationGroup] = useState<string>('');
+
+  useEffect(() => {
+    (async () => {
+      const metadata = await auth.getImpersonationMetadata();
+
+      setImpersonationUser(metadata?.user ?? '');
+      setImpersonationGroup(metadata?.groups?.[0] ?? '');
+    })();
+  }, [auth]);
+
+  const handleSetImpersonation = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    auth.setImpersonationMetadata({
+      user: impersonationUser,
+      groups: impersonationGroup ? [impersonationGroup] : undefined,
+    });
+
+    new FlashMessage(
+      'Impersonation configured successfully.',
+      messageType.SUCCESS,
+      messageTTL.MEDIUM
+    );
+  };
+
+  const handleClearImpersonation = () => {
+    auth.setImpersonationMetadata(null);
+
```

Comment:
Could we clear the form in the UI as well? 🙂

Reply:
Nah that would be too smart 😆